automated-testing/shadow-testing/readme.md

```diff
@@ -0,0 +1,61 @@
+There're some similarities with [Dark Launching](https://martinfowler.com/bliki/DarkLaunching.html), Dark Launching propose to integrate new feature into production release code and users are not able to see this feature, on the backend you can test your feature and improve the performance until it's acceptable. Also [Feature Toggle](https://martinfowler.com/bliki/FeatureToggle.html) approach is providing you an ability to enable/disable your new feature on the production on UI level, with this approach your new feature will be visible to users and you can collect feedback. Using Dark Launching with Feature Toggle approach can be very useful for introducing a new feature.
+
+## Applicable to
+
+- **Production deployments**: V-Next in Shadow testing always working separately and not effecting production. Users are not effected with this test.
+- **Infrastructure**: Shadow testing replicating the same traffic, in test environment you can have the same traffic on the production. It helps to produce real life test scenarios
+- **Handling Scale**: All traffic is replicated and you have a chance to see how your system scaling.
+
+## Shadow Testing Frameworks and Tools
+
+There are some tools to implement shadow testing. Main purpose of these tools is to compare responses of V-Current and V-Next then find the diffrences.
+
+- [Diffy](https://github.com/opendiffy/diffy)
+- [Envoy](https://www.envoyproxy.io)
+- [McRouter](https://github.com/facebook/mcrouter)
+- [Scientist](https://github.com/github/scientist)
+
+One of the most popular tool is [Diffy](https://github.com/opendiffy/diffy). It was created and used at Twitter then original author and a former Twitter employee maintains their own version of this project, called [Opendiffy](https://github.com/opendiffy/diffy). Twitter announced this tool on the blog as "[Testing services without writing tests](https://blog.twitter.com/engineering/en_us/a/2015/diffy-testing-services-without-writing-tests.html)".
+
+ As of today Diffy is used in production by Twitter, Airbnb, Baidu and Bytedance companies. It explains shadow testing feature like this:
+
+> Diffy finds potential bugs in your service using running instances of your new code and your old code side by side. Diffy behaves as a proxy and multicasts whatever requests it receives to each of the running instances. It then compares the responses, and reports any regressions that may surface from those comparisons. The premise for Diffy is that if two implementations of the service return “similar” responses for a sufficiently large and diverse set of requests, then the two implementations can be treated as equivalent and the newer implementation is regression-free.
+
+![Diffy Shadow Testing Architecture](images/diffy-shadow-testing.png)
+
+Diffy architecture
+
+## Conclusion
+
+Shadow Testing is a useful approach to replicate production environment in a test environment. In test environment main purpose is to compare results with production environment.
```

Comment:
I have difficulties understanding this sentence. Can you perhaps clarify a little what it means? I thought based on the above, that the main purpose of shadow testing was to being able to test items in production

Reply:
Thanks @TessFerrandez for your comments, I resolved all your comments. Please can you review the changes?